SAP Connector 5.2 (#1182)
mulesoft/docs-connectors: sap/5.2/modules/ROOT/pages/sap-connector-studio.adoc

= SAP Connector 5.2 Studio Configuration - Mule 4
:page-aliases: connectors::sap/sap-connector-studio.adoc

Install SAP Connector and configure the connector in Anypoint Studio.
First install the SAP Java Connector (JCo) libraries that you obtain from SAP and then you can create an app containing
one or more SAP Connector operations. Studio enables you to create an app and access the app from a browser.

To configure SAP Connector in Anypoint Studio:

. Add the connector to your Mule project.
. Configure the SAP connector.
. Configure an input source for the connector.

[[add-connector-to-project]]
== Add the Connector to Your Mule Project

Add SAP Connector to your Mule project to automatically populate the XML code with the connector's namespace and schema location and to add the required dependencies to the project's `pom.xml` file:

. In the *Mule Palette* view, click *(X) Search in Exchange*.

> . In the *Add Dependencies to Project* window, type `<connector name in lowercase>` in the search field.

Reviewer: ```suggestion
. In the *Add Dependencies to Project* window, type `sap` in the search field.
```

. Click *SP Connector* in *Available modules*.
. Click *Add*.
. Click *Finish*.

Adding a connector to a Mule project in Studio does not make that connector available to other projects in your Studio workspace.

== Select a Source Operation

> . In Anypoint Studio, click the *Mule Palette* and type "sap".

Reviewer: ```suggestion
. In Anypoint Studio, click the *Mule Palette* and type `sap`.
```

. Click the plus sign (*+*) on the right of the *Connector Configuration* field. +
This screen appears:
+
image::sap-connector-configuration-01.png[]

=== Install the IDOC Library

. Click *Add Dependency*.
. Click *Install*.
+
image::sap-connector-configuration-02.png[]
+
. In the *Install* menu, click *Browse* to locate the library file.
. Specify a unique group ID, artifact ID, and version.
+
image::sap-connector-configuration-02-in.png[]
+
. Click *Finish* or, if you want to cancel, click *Remove Dependency*.
+
image::sap-connector-configuration-02-finish.png[]

=== Install the JCO Library

. Click *Add Dependency*.
. Click *Install*.
+
image::sap-connector-configuration-02.png[]
+
. In the *Install* menu, click *Browse* to locate the library file.
. Specify a unique group ID, artifact ID, and version.
+
image::sap-connector-configuration-03-in.png[]
+
. Click *Finish* or, if you want to cancel, click *Remove Dependency*.
+
image::sap-connector-configuration-03-finish.png[]

=== Install the JCO Native Library

. Click *Add Dependency*.
. Click *Install*.
+
image::sap-connector-configuration-02.png[]
+
. In the *Install* menu, click *Browse* to locate the library file.
+

> On MacOS, starting with JCO version 3.1.2, the native library extension was changed from `jnilib` to `dylib`. Anypoint Studio versions 7.0 through 7.6 have a limitation on selecting the `dylib` file type.

Reviewer: I think we need to maybe reword this, something like: Starting with Anypoint Studio 7.0, there is a limitation..."
Because Studio is now on version 7.7. Maybe this limitation doesn't apply to 7.7? Needs to be verified with dev.

.. Select *Any*, and then select the `dylib` native library.
.. In the *Install file into local repository* window, enter values for the following fields:
* *Group ID*
* *Artifact ID*
* *Version*
+
. Specify a unique group ID, artifact ID, and version.
+
image::sap-connector-configuration-04-in.png[]
+
. Click *Finish* or, if you want to cancel, click *Remove Dependency*.
+
image::sap-connector-configuration-04-finish.png[]

When you are finished installing the libraries, the configuration screen appears as follows with green check marks for each successfully installed library:

image::sap-connector-libraries-installed.png[]

== Test the Connection

. Sign into your SAP login to get the information you need to configure the remaining *Global Element Properties* fields as described in <<gepparms,Configure in Studio>>.
. Click *Test Connection* to ensure that you have the correct login information.

== Configure the Connector

If you need more than one SAP Connector instance in your Mule project, you can create a global SAP element. The SAP Connector object holds the configuration properties that allow you to connect to the SAP server.

> To create a configuration for an SAP Connector, complete the following steps:

Reviewer: ```suggestion
To create a configuration for SAP Connector, complete the following steps:
```
